src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -356,9 +361,29 @@ public SegmentMetadata.ComponentMetadataMap writeData(IndexDescriptor indexDescr
                                                         ? x -> ordinalMap.inverse().getOrDefault(x, x)
                                                         : x -> x;
 
+            // Retrieve the first compressed vectors with more than MAX_PQ_TRAINING_SET_SIZE rows
+            var indexes = new ArrayList<>(indexContext.getView().getIndexes());
+            indexes.sort(Comparator.comparing(ssTableIndex -> ssTableIndex.getSSTable().descriptor.id));
+            CompressedVectors compressedVectors = null;
+            outer:
+            for (SSTableIndex index : indexes)
+            {
+                for (Segment segment : index.getSegments())
+                {
+                    if (segment.metadata.numRows < ProductQuantization.MAX_PQ_TRAINING_SET_SIZE)
+                        continue;
+
+                    var searcher = segment.getIndexSearcher();
+                    assert searcher instanceof CompressedVectorsSupplier;
```

Comment:
compressedVectors can be null, so since we need to handle that anyway it's probably simpler to just have the Hnsw code return null instead of adding a new interface here

Reply:
This is the index searcher, not the graph. I removed the interface in favor of `V2VectorIndexSearcher` since `V3VectorIndexSearcher` extends it for now.